unified-prototype/unified-plugin/plugin-android/src/main/java/org/gradle/api/experimental/android/AndroidTarget.java

```diff
@@ -3,27 +3,41 @@
 import com.android.build.api.dsl.BaseFlavor;
 import org.gradle.api.Action;
 import org.gradle.api.Named;
+import org.gradle.api.NonNullApi;
 import org.gradle.api.experimental.common.LibraryDependencies;
 import org.gradle.api.provider.Property;
 import org.gradle.api.tasks.Input;
 import org.gradle.api.tasks.Nested;
 
-public interface AndroidTarget extends Named {
+import javax.inject.Inject;
+
+@NonNullApi
+public abstract class AndroidTarget implements Named {
```

Comment:
If we didn't implement `Named`, I think we will be able to change all of these back into interfaces. 

I think that would clean up the the whole android extension, so we don't need to do any manual `objects.newInstance` calls.

That would make `getTarget` in the plugin more complex but I think that's a worthy trade. 

Thoughts?

Reply:
We ran out of time before I could ask in the meeting, but your first comment on the targets PR is something I was thinking about when coding and choosing this route.

I'm not concerned with the complexity of `getTarget()` in the plugin at all, but I was thinking that it makes sense that a target has a name, and we should preserve that.  I wasn't worried about the ugliness of the plugin implementation itself, with the `newInstance()` calls.